pkg/cri/server/image_pull.go

```diff
@@ -491,6 +491,7 @@ func newTransport() *http.Transport {
 		IdleConnTimeout:       30 * time.Second,
 		TLSHandshakeTimeout:   10 * time.Second,
 		ExpectContinueTimeout: 5 * time.Second,
+		TLSClientConfig:       &tls.Config{},
```

Comment:
```
	// TLSClientConfig specifies the TLS configuration to use with
	// tls.Client.
	// If nil, the default configuration is used.
	// If non-nil, HTTP/2 support may not be enabled by default.
	TLSClientConfig *tls.Config
```
nil has valid meaning here ^  

Where did the panic occur?

Reply:
OK. I see a couple locations where nil here would panic depending on particular config.. IMO we should fix those vs always have a null config and thus change behavior by removing use of the default configuration.

Reply:
How about this change?  as before.

```golang
			var (
				transport = newTransport()
				client    = &http.Client{Transport: transport}
				config    = c.config.Registry.Configs[u.Host]
			)
			if config.TLS != nil {
				transport.TLSClientConfig, err = c.getTLSConfig(*config.TLS)
				if err != nil {
					return nil, errors.Wrapf(err, "get TLSConfig for registry %q", e)
				}
			} else if isLocalHost(host) && u.Scheme == "http" {
				// Skipping TLS verification for localhost
                                 transport.TLSClientConfig = &tls.Config{}
				transport.TLSClientConfig.InsecureSkipVerify = true
			}
```

Reply:
or.. 
```
transport.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
```

Reply:
fixed